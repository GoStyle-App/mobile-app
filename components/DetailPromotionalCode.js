import * as React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { Colors, IconButton } from 'react-native-paper';
import getDateText from './../filters/getDateText';

export default function DetailPromotionalCode({ route, navigation }) {
    const { code } = route.params;

    return (
        <View style={ styles.container }>
            <Text style={ styles.code }>{ code.code }</Text>
            <Text style={ styles.date }>{ getDateText(code) }</Text>
            <Text style={ styles.label }>{ code.label }</Text>
            <Text style={ styles.description }>{ code.description }</Text>
            <View style={ styles.buttonContainer }>
                <IconButton
                    icon="arrow-left"
                    color={ Colors.white }
                    size={ 35 }
                    onPress={() => {
                        navigation.navigate('Home');
                    }}
                />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        textAlign: 'center',
        alignItems: 'center',
        marginTop: 30,
    },
    code: {
        color: '#FFF',
        backgroundColor: '#000',
        fontSize: 38,
        width: 200,
        marginBottom: 18,
        padding: 10,
        textAlign: 'center',
    },
    date: {
        color: '#38B6FF',
        marginBottom: 80,
    },
    label: {
        fontSize: 16,
        textTransform: 'uppercase',
        marginBottom: 20,
    },
    description: {
        color: '#737373',
        paddingLeft: 40,
        paddingRight: 40,
        textAlign: 'justify',
    },
    buttonContainer: {
        backgroundColor: '#38B6FF',
        borderRadius: 50,
        marginRight: 250,
        marginTop: 275,
    }
});
