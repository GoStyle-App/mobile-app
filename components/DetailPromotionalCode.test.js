import React from 'react';
import DetailPromotionalCode from '../components/DetailPromotionalCode';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

describe('Display corresponding promotional code.', () => {
    test('Correctly render detail of promotional code', () => {
        const code = {
            "code": "ETE2020",
            "description": "Bénéficier de 20% sur vos achats cet été",
            "endDate": "2021-05-30T00:00:00+00:00",
            "id": 1,
            "label": "Promotion été 2020",
            "startDate": "2021-04-24T00:00:00+00:00",
        };
        const route = {
            "key": "DetailPromotionalCode-6AZICdFaVndzfzPFdRhy4",
            "name": "DetailPromotionalCode",
            "params": {
                "code": code,
            },
        };
        const navigation = {
            "addListener": function addListener(){},
            "canGoBack": function canGoBack(){},
            "closeDrawer": function anonymous(){},
            "dangerouslyGetParent": function dangerouslyGetParent(){},
            "dangerouslyGetState": function anonymous(){},
            "dispatch": function dispatch(){},
            "goBack": function anonymous(){},
            "isFocused": function isFocused(){},
            "jumpTo": function anonymous(){},
            "navigate": function anonymous(){},
            "openDrawer": function anonymous(){},
            "removeListener": function removeListener(){},
            "reset": function anonymous(){},
            "setOptions": function setOptions(){},
            "setParams": function anonymous(){},
            "toggleDrawer": function anonymous(){},
        };
        const component = (
            <NavigationContainer>
                <DetailPromotionalCode navigation={navigation} route={route} />
            </NavigationContainer>
        );
        const detailPromotionalComponent = render(component);

        expect(detailPromotionalComponent).toMatchSnapshot();
    });
});
