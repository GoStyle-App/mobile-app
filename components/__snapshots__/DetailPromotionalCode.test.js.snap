// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display corresponding promotional code. Correctly render detail of promotional code 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 30,
      "textAlign": "center",
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#000",
        "color": "#FFF",
        "fontSize": 38,
        "marginBottom": 18,
        "padding": 10,
        "textAlign": "center",
        "width": 200,
      }
    }
  >
    ETE2020
  </Text>
  <Text
    style={
      Object {
        "color": "#38B6FF",
        "marginBottom": 80,
      }
    }
  >
    À partir du 24/04/2021 Jusqu'au 30/05/2021
  </Text>
  <Text
    style={
      Object {
        "fontSize": 16,
        "marginBottom": 20,
        "textTransform": "uppercase",
      }
    }
  >
    Promotion été 2020
  </Text>
  <Text
    style={
      Object {
        "color": "#737373",
        "paddingLeft": 40,
        "paddingRight": 40,
        "textAlign": "justify",
      }
    }
  >
    Bénéficier de 20% sur vos achats cet été
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#38B6FF",
        "borderRadius": 50,
        "marginRight": 250,
        "marginTop": 275,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 26.25,
              "height": 52.5,
              "width": 52.5,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      <View>
        <Text />
      </View>
    </View>
  </View>
</View>
`;
